### DISCUSSION TITLE
Instantiate이후에 스크립트가 비활성화 됩니다

### DISCUSSION THREAD

**Discussion by PH62mem** [naverz/zepeto-studio-kor#803] [5. Q&A]
유니티에서
var obj = Object.Instantiate(this.sco) as GameObject;
로 이 오브젝트를 복사해 왔는데요
아래 보시는 것처럼 스크립트가 꺼져서 복사됩니다.
![image](https://user-images.githubusercontent.com/105163890/170319735-6de791c4-2003-4163-b773-467154b393f0.png)
![image](https://user-images.githubusercontent.com/105163890/170319745-3d7c0fd7-66f4-4724-b79f-22514118ea18.png)

해결방법이 있을까요?

**Comment by Zepeto-TechEvangelist** [comment]
안녕하세요, 전체 코드 및 프리팹 경로 스크린샷 등을 올려주시면 도움을 드릴 수 있을 것 같습니다.

**Reply by PH62mem** [reply]
![image](https://user-images.githubusercontent.com/105163890/170516096-89850ff3-1c14-4896-800c-29412efee7d8.png)
![image](https://user-images.githubusercontent.com/105163890/170515796-78201008-3d16-4c26-98cf-890d1d5c60d2.png)
![image](https://user-images.githubusercontent.com/105163890/170515705-bd0f20fa-298e-4e5b-997f-c4708907e670.png)
![image](https://user-images.githubusercontent.com/105163890/170515751-aa8bacdd-801e-4f4b-8b76-47d9a7d06128.png)

prefab였는데 계속 스크립트랑 연결이 끊겨서 오브젝트로 나뒀습니다.
(Ext component ... 부분에서 꼐속 UI에대한 연결이 끊혀서)

**Reply by x4cf** [reply]
OnTriggerEnter 마지막줄에 var obj.GetComponent<GetItem>().enable = true;
를 넣어보는건 어떨까요?

**Reply by PH62mem** [reply]
> OnTriggerEnter 마지막줄에 var obj.GetComponent().enable = true; 를 넣어보는건 어떨까요?

![image](https://user-images.githubusercontent.com/105163890/170828614-df2aa08d-32d9-472a-948e-8a093867848e.png)
오류가 생깁니다.

**Comment by Zepeto-TechEvangelist** [comment]
자기 자신이 파괴되면서 오브젝트를 생성해서 그런것 같습니다.
오브젝트를 생성하는 스크립트를 별도로 만들어서 관리할 것을 추천 드립니다.

**Reply by PH62mem** [reply]
![image](https://user-images.githubusercontent.com/105163890/170828642-1f46b992-ecf3-4a43-bf44-4b4cfb02afd7.png)
![image](https://user-images.githubusercontent.com/105163890/170828646-c0209e1e-f505-47ca-b35c-7ccc303c580a.png)
![image](https://user-images.githubusercontent.com/105163890/170829023-927c44b8-b5be-4474-a89b-2b0383a91f21.png)
![image](https://user-images.githubusercontent.com/105163890/170829037-16ef3a1b-ee41-4d17-85a3-53e6fe90ac73.png)
![image](https://user-images.githubusercontent.com/105163890/170829046-69895c0a-bcfc-47fb-a68c-73e23401ace6.png)
오브젝트를 생성하는 스크립트를 별도로 만들어서 스크립트 체크가 풀리던 문제는 해결되는걸 확인했습니다. Prefab으로 불러오면 스크립트가 꺼지지는 않습니다만...
start로 만든 prefab 첫번쨰 복사 오브젝트 말고는 저렇게 연결이 끊어져서 제 기능을 하지 못합니다....

**Reply by Zepeto-TechEvangelist** [reply]
연결이 끊어지는 부분은 내부에서 확인해 보겠습니다.

**Reply by Zepeto-TechEvangelist** [reply]
안녕하세요, 연결이 끊기는 상황은 일반적이지는 않은것 같습니다.
아래 부분 확인 부탁 드립니다.
1. 인스턴스로 생성하는 프리팹 경로가 Project에 있고, 그 프리팹을 끌어다 놓았는지 확인 부탁 드립니다. ( Hierarchy에 있는 프리팹의 경우 끊길 수 있습니다)
2. Setbox호출 스크립트를 파괴 이전으로 순서를 바꿔보신 후 문제가 없는지 확인 부탁 드립니다.

위 방법으로도 해결되지 않는다면 프로젝트 파일 첨부 부탁 드립니다~
